### DISCUSSION TITLE
InjectedProviders collision

### DISCUSSION THREAD

**Discussion by reddigszymon** [WalletConnect/walletconnect-monorepo#3924] [Developer Support] [answered]
I am aware that there is currently a bug, where if someone connects with both Coinbase Wallet + Metamask it will sometimes open up both extensions for example when paying.

In order to bypass that, I need to know which wallet is currently chosen by the user - coinbase wallet or metamask, so i can perform some magic on the window.ethereum object and choose the correct provider when for example paying for something.

Does the web3modal give any way of detecting which wallet is currently connected (metamask, coinbase wallet, or something else)?

**Comment by glitch-txs** [comment]
Hi, yes we have fixed this issue by implementing [EIP-6963](https://eips.ethereum.org/EIPS/eip-6963) in Web3Modal, these wallets we'll be detected when installed and there won't be any conficts when connecting.

You can test it here: https://web3modal.com/
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/66949816/76f0d252-f114-4620-b772-81fd6c721560)

**Reply by reddigszymon** [reply]
So when for example paying for something can i use window.ethereum for the provider?

Because right now if i connect with Metamask, then i switch to coinbase, when i try to pay still metamask opens up. I have the latest web3modal version.

**Reply by glitch-txs** [reply]
can you share your configuration?

**Reply by reddigszymon** [reply]
const provider = new ethers.BrowserProvider(window.ethereum);
const signer = await provider.getSigner();

then i create the contract instance, and just call the transfer method.

Right now when i connect with metamask -> disconnect it from within the application (not metamask itself), i switch to coinbase, run the payment function, metamask opens up, not coinbase.

**Reply by glitch-txs** [reply, marked as answer]
dont use window.ethereum

you can get the provider and signer from hooks/actions

https://docs.walletconnect.com/web3modal/javascript/actions#subscribe-to-theme-changes

**Comment by reddigszymon** [comment]
> Hi, yes we have fixed this issue by implementing [EIP-6963](https://eips.ethereum.org/EIPS/eip-6963) in Web3Modal, these wallets we'll be detected when installed and there won't be any conficts when connecting.
> 
> You can test it here: https://web3modal.com/ ![image](https://user-images.githubusercontent.com/66949816/282756644-76f0d252-f114-4620-b772-81fd6c721560.png)



> can you share your configuration?
